ext/RMagick/rmagick.c

```diff
@@ -414,3 +414,53 @@ Magick_set_log_format(VALUE class, VALUE format)
     return class;
 }
 
+/**
+ * Get the ImageMagick delegate information.
+ * 
+ * Ruby usage:
+ *   - @verbatim Magick.delegates @endverbatim
+ *
+ * Notes:
+ *   The return hash value constains `decode` and `encode` key.
+ *
+ * @return the hash.
+ */
+
+VALUE
+Magick_delegates(VALUE class)
+{
+    const DelegateInfo **delegate_info;
+    ExceptionInfo *exception;
+    size_t number_delegates;
+    size_t i;
+    VALUE hash;
+    VALUE ary_decode;
+    VALUE ary_encode;
+
+    exception = AcquireExceptionInfo();
+    delegate_info = GetDelegateInfoList("*", &number_delegates, exception);
+    CHECK_EXCEPTION();
+    DestroyExceptionInfo(exception);
+
+    ary_decode = rb_ary_new2(number_delegates);
+    ary_encode = rb_ary_new2(number_delegates);
```

Comment:
Maybe this doesn't matter, but it looks like we're potentially allocating a longer array than we will need here, since not all delegates will necessarily be `decode` and `encode` both. For example, if we have 1 delegate that is `decode` and 1 that is `encode`, we'll end up allocating 2 slots for each array, when it should be 1 for each.

Is that something worth worrying about? I guess the code will be much more complicated if we try to handle this.

Reply:
Make sense. I will fix the code.